src/doc/trpl/advanced-linking.md

```diff
@@ -0,0 +1,175 @@
+% Advanced Linking
+
+The common cases of linking with Rust have been covered earlier in this book,
+but supporting the range of linking possibilities made available by other
+languages is important for Rust to achieve seamless interaction with native
+libraries.
+
+# Native libraries required by upstream crates
```

Comment:
I think that something like this may best appropriately live in a reference-style piece of documentation rather than tutorial-like docs like the book.